### ISSUE TITLE
Documentation for matrix path variables

### ISSUE THREAD

**Issue by leeshangqian** [http4s/http4s#5006]
There doesn't seem to be any documentation on https://http4s.org/v0.23/dsl/ regarding how to work with matrix path variables.

Matrix path variables are supported as part of https://github.com/http4s/http4s/blob/main/dsl/src/main/scala/org/http4s/dsl/impl/Path.scala#L157.

I'd like to help if there isn't anyone else working on this.

**Comment by rossabaker** [comment]
I don't believe there is, and the help would be much appreciated!  I'd open the PR against the series/0.22 branch, because they should work exactly the same way, and then your work would merge up so it's in 0.23 and beyond as well.

Thanks!
